### DISCUSSION TITLE
How to deal with a targetted update scenario?

### DISCUSSION THREAD

**Discussion by Erwyn** [containrrr/watchtower#1244] [Q&A] [answered]
Hi there, 

I'm new to Watchtower and I'm trying to wrap my head around my 3 scenarios and how to make them work with Watchtower. I have a fairly banal setup using docker-compose and I was considering using Watchtower to introduce Automatic updates in there. I think I have three scenarios:

1. Daily check of new images for all the images, basically the default behaviour of Watchtower if I read correctly
2. Manually trigger the update of all containers, which could be done using the HTTP API with periodical polls still activated to keep `1.` in check.
3. Manually trigger the update of one specific container, which is not possible via HTTP API without spawning one Watchtower per-container

How could I deal with this? Or is this even a proper Watchtower scenario?

I imagined using the HTTP API but as said in `3.` it does not seem to support a parameter to specify which container(s) to target on this update. I also thought I could have one Watchtower container running for the `1.` and `2.` using HTTP API and periodic polls and use a RUN ONCE Watchtower when I want to update one container specifically, but if I get it correctly it would kill my initial container so I would have to start it back again afterwards.

Am I missing something? Does a strategy exists to target one container for update?

Thanks!

**Comment by piksel** [comment, marked as answer]
No, this is fairly accurate. 
It *should* be possible to run it, activating both the HTTP API and run the schedule, but it was implemented with the assumption that those should be exclusive.
There is currently no params you can pass to the update endpoint, but specifying the filters seems like a good idea.

**Reply by Erwyn** [reply]
Okay, so if I get this correctly, the correct way of dealing with this would be to add the possibility to pass the filters as parameters to the HTTP API so that I can target some containers ? More than the RUN ONCE scenario where I would need to restart the other Watchtower container afterwards ?

**Reply by piksel** [reply]
Pretty much yeah. Watchtower was both designed to only run a single instance on one machine, and to explicitly only work in one of many modes. There has been discussions about how this should be rewritten, but no work on implementing this has been done yet afaik.

**Reply by Erwyn** [reply]
Ok, thank you very much !
